src/EnergyPlus/HVACManager.cc

```diff
@@ -2829,107 +2687,101 @@ void ReportAirHeatBalance(EnergyPlusData &state)
             if (state.dataHeatBal->ZoneAirBalance(j).BalanceMethod == DataHeatBalance::AirBalance::Quadrature &&
                 ZoneLoop == state.dataHeatBal->ZoneAirBalance(j).ZonePtr) {
-                if (thisZoneHB.MAT > Zone(ZoneLoop).OutDryBulbTemp) {
-                    ZnAirRpt(ZoneLoop).OABalanceHeatLoss =
-                        thisZoneHB.MDotCPOA * (thisZoneHB.MAT - Zone(ZoneLoop).OutDryBulbTemp) * TimeStepSysSec * ADSCorrectionFactor;
-                    ZnAirRpt(ZoneLoop).OABalanceHeatGain = 0.0;
+                if (thisZoneHB.MAT > zone.OutDryBulbTemp) {
+                    znAirRpt.OABalanceHeatLoss = thisZoneHB.MDotCPOA * (thisZoneHB.MAT - zone.OutDryBulbTemp) * TimeStepSysSec * ADSCorrectionFactor;
+                    znAirRpt.OABalanceHeatGain = 0.0;
                 } else {
-                    ZnAirRpt(ZoneLoop).OABalanceHeatLoss = 0.0;
-                    ZnAirRpt(ZoneLoop).OABalanceHeatGain =
-                        -thisZoneHB.MDotCPOA * (thisZoneHB.MAT - Zone(ZoneLoop).OutDryBulbTemp) * TimeStepSysSec * ADSCorrectionFactor;
+                    znAirRpt.OABalanceHeatLoss = 0.0;
+                    znAirRpt.OABalanceHeatGain = -thisZoneHB.MDotCPOA * (thisZoneHB.MAT - zone.OutDryBulbTemp) * TimeStepSysSec * ADSCorrectionFactor;
                 }
-                H2OHtOfVap = PsyHgAirFnWTdb(state.dataEnvrn->OutHumRat, Zone(ZoneLoop).OutDryBulbTemp);
+                H2OHtOfVap = Psychrometrics::PsyHgAirFnWTdb(state.dataEnvrn->OutHumRat, zone.OutDryBulbTemp);
                 if (thisZoneHB.ZoneAirHumRat > state.dataEnvrn->OutHumRat) {
-                    ZnAirRpt(ZoneLoop).OABalanceLatentLoss = thisZoneHB.MDotOA * (thisZoneHB.ZoneAirHumRat - state.dataEnvrn->OutHumRat) *
-                                                             H2OHtOfVap * TimeStepSysSec * ADSCorrectionFactor;
-                    ZnAirRpt(ZoneLoop).OABalanceLatentGain = 0.0;
-                } else if (thisZoneHB.ZoneAirHumRat <= state.dataEnvrn->OutHumRat) {
-                    ZnAirRpt(ZoneLoop).OABalanceLatentGain = thisZoneHB.MDotOA * (state.dataEnvrn->OutHumRat - thisZoneHB.ZoneAirHumRat) *
-                                                             H2OHtOfVap * TimeStepSysSec * ADSCorrectionFactor;
-                    ZnAirRpt(ZoneLoop).OABalanceLatentLoss = 0.0;
+                    znAirRpt.OABalanceLatentLoss = thisZoneHB.MDotOA * (thisZoneHB.ZoneAirHumRat - state.dataEnvrn->OutHumRat) * H2OHtOfVap *
+                                                   TimeStepSysSec * ADSCorrectionFactor;
+                    znAirRpt.OABalanceLatentGain = 0.0;
+                } else {
+                    znAirRpt.OABalanceLatentGain = thisZoneHB.MDotOA * (state.dataEnvrn->OutHumRat - thisZoneHB.ZoneAirHumRat) * H2OHtOfVap *
+                                                   TimeStepSysSec * ADSCorrectionFactor;
+                    znAirRpt.OABalanceLatentLoss = 0.0;
```

Comment:
[src/EnergyPlus/HVACManager.cc:2876]:(style),[multiCondition],Expression is always true because 'else if' condition is opposite to previous condition at line 2872.